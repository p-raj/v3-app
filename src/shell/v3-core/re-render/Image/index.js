import React from 'react';
import PropTypes from 'prop-types';
import { Image as RNImage, TouchableOpacity } from 'react-native';


class Image extends React.Component {
    render() {
        return (
            <TouchableOpacity onPress={this.props.onPress}>
                <RNImage {...this.props}/>
            </TouchableOpacity>
        )
    }
}

Image.propTypes = {
    ...RNImage.propTypes,
    onPress: PropTypes.func
};

export default Image;
