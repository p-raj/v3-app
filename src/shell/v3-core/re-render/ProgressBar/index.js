import PropTypes from 'prop-types';
import React from 'react';
import { ProgressBar as PB } from 'react-native-web';

export default class ProgressBar extends React.Component {
    render() {
        return (
            <PB
                style={{width: '100%'}}
                color={this.props.color}
                indeterminate={this.props.indeterminate}
                progress={this.props.progress}
                trackColor={this.props.trackColor}/>
        )
    }
}

ProgressBar.defaultProps = {
    trackColor: '#e0e0e0'
};

ProgressBar.propTypes = {
    color: PropTypes.string,
    indeterminate: PropTypes.bool,
    progress: PropTypes.number,
    trackColor: PropTypes.string,
};