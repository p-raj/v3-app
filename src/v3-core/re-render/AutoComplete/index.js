import React from 'react';
import PropTypes from 'prop-types';
import { ListView, Text, TextInput, TouchableOpacity, View } from 'react-native';

const styles = {
    container: {
        width: 200,
    },
    autoCompleteInput: {
        width: 200,
        height: 40,
    },
    listContainer: {
        zIndex: 1,
        position: 'absolute',
        marginTop: 40,
        backgroundColor: '#FFF',
        width: 'inherit'
    },
    listItem: {
        height: 40,
        justifyContent: 'center',
        borderWidth: 2,
        borderStyle: 'solid',
        borderColor: '#e0e0e0',
        borderTopWidth: 0,
    },
};

export default class AutoComplete extends React.Component {
    constructor(props) {
        super(props);

        const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
        this.state = {
            allData: props.list,
            dataSource: ds.cloneWithRows([]),
            query: '',
            showList: false,
        };
    }

    render() {
        return (
            <View style={styles.container}>
                <TextInput style={styles.autoCompleteInput}
                           placeholder={this.props.placeholder}
                           value={this.state.query}
                           onBlur={() => this.setState({showList: false})}
                           onChangeText={this.onQueryChange}/>
                {this.state.showList
                &&
                <View style={styles.listContainer}>
                    <ListView
                        enableEmptySections={true}
                        dataSource={this.state.dataSource}
                        renderRow={
                            (rowData) => (
                                <TouchableOpacity
                                    style={styles.listItem}
                                    onPress={
                                        () => {
                                            this.setState({
                                                showList: false, query: rowData
                                            }, () => {
                                                if (this.props.onItemSelected)
                                                    this.props.onItemSelected(this.state.query);
                                            });
                                        }}>
                                    <Text style={{marginLeft: 10}}>
                                        {rowData}
                                    </Text>
                                </TouchableOpacity>
                            )
                        }
                    />
                </View>
                }
            </View>
        )
    }

    onQueryChange = (value) => {
        this.setState({query: value});
        this.filter(value);
    };

    filter = (query) => {
        query = query.trim();
        let filteredData = this.state.allData.filter((item) => {
            if (this.props.ignoreCase)
                return item.toLowerCase().includes(query.toLowerCase());

            return item.includes(query);
        });

        const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
        this.setState({
                dataSource: ds.cloneWithRows(filteredData),
                showList: true
            }
        );
    };
}

AutoComplete.defaultProps = {
    ignoreCase: false,
};

AutoComplete.propTypes = {
    placeholder: PropTypes.string,
    list: PropTypes.array.isRequired,
    ignoreCase: PropTypes.bool,
    onItemSelected: PropTypes.func,
};
