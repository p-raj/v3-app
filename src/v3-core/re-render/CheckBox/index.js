import React from 'react';
import PropTypes from 'prop-types';
import { CheckBox as CB } from 'react-native-elements';


export default class CheckBox extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            checked: this.props.checked
        }
    }

    render() {
        return (
            <CB
                title={this.props.title}
                checked={this.state.checked}
                checkedIcon={'check-square'}
                checkedColor={this.props.checkboxColor}
                uncheckedColor={this.props.checkboxColor}
                onPress={this.onClick}
            />
        )
    }

    onClick = () => {
        this.setState({
            checked: !this.state.checked
        }, this.afterSetStateCallback);
    };

    afterSetStateCallback = () => {
        if (typeof this.props.onCheckedChange === 'function')
            this.props.onCheckedChange(this.state.checked)
    }
}

CheckBox.defaultProps = {
    checked: false,
    checkboxColor: 'black',
};

CheckBox.propTypes = {
    title: PropTypes.string.isRequired,
    checked: PropTypes.bool,
    onCheckedChange: PropTypes.func,
    checkboxColor: PropTypes.string,
};