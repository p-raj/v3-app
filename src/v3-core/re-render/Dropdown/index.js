import React from 'react';
import PropTypes from 'prop-types';
import { Text, TouchableOpacity, View } from 'react-native';
import RecyclerView from '../RecyclerView';
import Icon from '../Icon';

const styles = {
    container: {
        width: 200,
    },
    autoCompleteInput: {
        width: 200,
        height: 40,
        justifyContent: 'center',
        borderWidth: 2,
        borderStyle: 'solid',
        borderColor: '#e0e0e0',
    },
    listContainer: {
        zIndex: 1,
        position: 'absolute',
        marginTop: 40,
        backgroundColor: '#FFF',
        width: 200
    },
    listItem: {
        height: 40,
        justifyContent: 'center',
        borderWidth: 2,
        borderStyle: 'solid',
        borderColor: '#e0e0e0',
        borderTopWidth: 0,
    },
};

export default class Dropdown extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            showList: false,
            selectedItem: ''
        };
    }

    render() {
        return (
            <View style={styles.container}>
                <TouchableOpacity
                    style={[styles.autoCompleteInput, {backgroundColor: this.props.placeholderBackground}]}
                    onPress={() => {
                        this.setState({showList: !this.state.showList})
                    }}>
                    <View
                        style={[{flexDirection: 'row', marginRight: 10, marginLeft: 10}]}>
                        <Text style={{flex: 1, color: this.props.placeholderColor}}>
                            {this.state.selectedItem.value || this.props.placeholder}
                        </Text>
                        {(!this.state.selectedItem) ?
                            <Icon
                                name={this.state.showList ? 'angle-up' : 'angle-down'}
                                type='font-awesome'
                                color={this.props.placeholderColor}
                            /> :
                            <Icon
                                name='close'
                                type='font-awesome'
                                color={this.props.placeholderColor}
                                onPress={() => {
                                    this.onItemClicked('')
                                }}
                            />
                        }
                    </View>
                </TouchableOpacity>

                {
                    this.state.showList
                    &&
                    <View style={styles.listContainer}>
                        <RecyclerView
                            onItemClicked={this.onItemClicked}
                            style={{maxHeight: this.props.listHeight}}
                            enableEmptySections={true}
                            dataSource={this.props.list}
                            renderRow={(rowData) => (
                                <TouchableOpacity
                                    style={[styles.listItem, this.props.itemStyle]}>
                                    <Text style={{marginLeft: 10}}>
                                        {rowData.value}
                                    </Text>
                                </TouchableOpacity>
                            )}
                        />
                    </View>
                }

            </View>
        )
    }

    onItemClicked = (rowData) => {
        this.setState({showList: false, selectedItem: rowData});
        if (this.props.onItemSelected)
            this.props.onItemSelected(rowData);
    }
}

Dropdown.defaultProps = {
    listHeight: 200,
    placeholder: 'Select',
    placeholderColor: 'black'
};

Dropdown.propTypes = {
    list: PropTypes.arrayOf(PropTypes.shape({
        key: PropTypes.number.isRequired,
        value: PropTypes.string.isRequired,
    })).isRequired,
    placeholder: PropTypes.string,
    onItemSelected: PropTypes.func,
    listHeight: PropTypes.number,
    placeholderBackground: PropTypes.string,
    placeholderColor: PropTypes.string,
    itemStyle: PropTypes.object,
};
