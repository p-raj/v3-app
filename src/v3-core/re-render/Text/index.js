import React from 'react';
import { Text as RNText } from 'react-native';

class Text extends React.Component {
    render() {
        return (
            <RNText {...this.props}/>
        )
    }
}

Text.propTypes = RNText.propTypes;

export default Text;
