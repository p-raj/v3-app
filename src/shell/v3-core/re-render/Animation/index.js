import React from 'react';
import View from '../../components/ui/View';
import Text from '../Text/index';


export default () => {
    return (
        <View>
            <Text>
                The component is not available on Web.
            </Text>
        </View>
    );
};