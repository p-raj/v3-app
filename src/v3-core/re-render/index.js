import Animation from './Animation';
import AutoComplete from './AutoComplete';
import Button from './Button';
import CheckBox from './CheckBox';
import Dropdown from './Dropdown';
import Icon from './Icon';
import Image from './Image';
import Text from './Text';
import MultiSelect from './MultiSelect';
import { MultiSelectItem } from './MultiSelect';
import ProgressBar from './ProgressBar';
import RadioGroup from './Radio/RadioGroup';
import RadioButton from './Radio/RadioButton';
import RecyclerView from './RecyclerView';
import Spinner from './Spinner';
import TextArea from './TextArea';
import TextInput from './TextInput';

export {
    Animation, AutoComplete, Button, CheckBox, Dropdown,
    Icon, Image, Text, MultiSelect,
    MultiSelectItem, ProgressBar, RadioGroup, RadioButton,
    RecyclerView, Spinner, TextArea, TextInput
 };