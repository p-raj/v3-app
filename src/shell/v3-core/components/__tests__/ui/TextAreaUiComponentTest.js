import React from 'react';
import renderer from 'react-test-renderer';
import { TextArea } from 're-render';


describe('Suite for testing textArea:', () => {
    it('renders a textArea as expected (matches the snapshot)', () => {
        const title = "Label";
        let onPress = () => {
            console.log("Label pressed")
        };
        const TextAreaShot = renderer.create(
            <TextArea
                value={title}
                onPress={onPress}
            />
        );
        expect(TextAreaShot).toMatchSnapshot();
    });
});

