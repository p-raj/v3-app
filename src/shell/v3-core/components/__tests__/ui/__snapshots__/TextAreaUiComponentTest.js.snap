exports[`Suite for testing textArea: renders a textArea as expected (matches the snapshot) 1`] = `
<TextInput
  onPress={[Function]}
  style={Object {}}
  underlineColorAndroid="transparent"
  value="Label" />
`;
